### PR TITLE
Upgrade to angulillos-0.6

### PR DESCRIPTION
**Pull request by laughedelic**

### CODE REVIEW

**Comment by laughedelic** [comment]
I have a doubt about `query` in the indexes. I have removed the compare-predicate parameter in [bio4j/angulillos#36](https://github.com/bio4j/angulillos/pull/36/files#diff-c529dc44e6b687d45819d5e3048b61cdL21), because I thought that it was used only internally in the `Unique` and `List` sub-interfaces. 

But here I see that it was a general method, to allow angulillos users to query any index with a given predicate. Now if you are using bio4j-core writing backend-independent code, you can't do that.

So I guess, I need to return that back in angulillos in some form, but without using [`blueprints.Compare`](http://www.tinkerpop.com/docs/javadocs/blueprints/2.5.0/com/tinkerpop/blueprints/Compare.html).

**Comment by eparejatobes** [comment]
I vote for depending on TinkerPop 3.x

File: src/main/java/com/bio4j/angulillos/titan/TitanUntypedGraph.java [hunk 1]
```diff
@@ -206,14 +206,30 @@ EdgeLabelMaker titanLabelMakerForEdgeType(TitanManagement mgmt, RT relationshipT
...
+      case atLeastOneToOne:        labelMaker.multiplicity(Multiplicity.MANY2ONE);
+      case atLeastOneToAtMostOne:  labelMaker.multiplicity(Multiplicity.MANY2ONE);
+      case atLeastOneToAtLeastOne: labelMaker.multiplicity(Multiplicity.MULTI);
+      case atLeastOneToAny:        labelMaker.multiplicity(Multiplicity.MULTI);
+
+      case anyToOne:               labelMaker.multiplicity(Multiplicity.MANY2ONE);
+      case anyToAtMostOne:         labelMaker.multiplicity(Multiplicity.MANY2ONE);
+      case anyToAtLeastOne:        labelMaker.multiplicity(Multiplicity.MULTI);
+      case anyToAny:               labelMaker.multiplicity(Multiplicity.MULTI);
+    }
```
Comment (laughedelic) [inline src/main/java/com/bio4j/angulillos/titan/TitanUntypedGraph.java]: @eparejatobes this needs your review

Comment (laughedelic) [inline src/main/java/com/bio4j/angulillos/titan/TitanUntypedGraph.java]: commenting on a particular commit? aha.. 😒
this comment was for the arities commit.

File: src/main/java/com/bio4j/angulillos/titan/TitanUntypedGraph.java [hunk 2]
```diff
@@ -234,12 +206,29 @@ EdgeLabelMaker titanLabelMakerForEdgeType(TitanManagement mgmt, RT relationshipT
...
+
+      case atLeastOneToOne:        labelMaker.multiplicity(Multiplicity.MANY2ONE);
+      case atLeastOneToAtMostOne:  labelMaker.multiplicity(Multiplicity.MANY2ONE);
+      case atLeastOneToAtLeastOne: labelMaker.multiplicity(Multiplicity.MULTI);
+      case atLeastOneToAny:        labelMaker.multiplicity(Multiplicity.MULTI);
+
+      case anyToOne:               labelMaker.multiplicity(Multiplicity.MANY2ONE);
+      case anyToAtMostOne:         labelMaker.multiplicity(Multiplicity.MANY2ONE);
+      case anyToAtLeastOne:        labelMaker.multiplicity(Multiplicity.MULTI);
+      case anyToAny:               labelMaker.multiplicity(Multiplicity.MULTI);
```
Comment (laughedelic) [inline src/main/java/com/bio4j/angulillos/titan/TitanUntypedGraph.java]: @eparejatobes here. this needs a review.

File: src/main/java/com/bio4j/angulillos/titan/TitanUntypedGraph.java (hunk 1 above)
Comment (eparejatobes) [inline src/main/java/com/bio4j/angulillos/titan/TitanUntypedGraph.java]: @laughedelic what's going on here in this conversation between you and @laughedelic ?

File: src/main/java/com/bio4j/angulillos/titan/TitanUntypedGraph.java (hunk 2 above)
Comment (eparejatobes) [inline src/main/java/com/bio4j/angulillos/titan/TitanUntypedGraph.java]: OK, will do

**Comment by laughedelic** [comment]
This is done. I'm leaving this PR for review. Meanwhile I'll start working on #16, based on this branch.

File: src/main/java/com/bio4j/angulillos/titan/TitanUntypedGraph.java (hunk 2 above)
Comment (eparejatobes) [inline src/main/java/com/bio4j/angulillos/titan/TitanUntypedGraph.java]: :+1:

**Pull request merged**